error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/StreamRules.java

```diff
@@ -167,6 +167,34 @@ Stream<R> after(Stream<T> stream, Function<? super S, ? extends Stream<? extends
     }
   }
 
+  /** Apply filtering before sorting to reduce the number of elements to sort. */
```

Comment:
Perhaps slightly more idiomatic:
```suggestion
  /** Filter before sorting to reduce the number of elements to sort. */
```

Reply:
Alternatively:
```suggestion
  /** Apply {@link Stream#filter(Predicate)} before {@link Stream#sorted()} to reduce the number of elements to sort. */
```

In most cases we do reference the relevant methods, so let's go with this one.